### ISSUE TITLE
full text search option minimum_should_match 

### ISSUE THREAD

**Issue by joalmjoalm** [crate/crate#4183]
Question:

I try using below full text search clause with `minimum_should_match` but I dont get it to work with numbers of `minimum_should_match >1`

```sql
select name from TableX where match((name_ft), 'aaa bbb') using best_fields with (minimum_should_match=2) order by _score desc
```

where 'aaa' and 'bbb' are some strings.

`TableX` columns:
```
 "name" STRING,
 INDEX "name_ft" USING FULLTEXT ("name") WITH (analyzer = 'standard' )
```

Above select with min = 2 also match on only 'aaa'/'bbb' in name_ft.

running docker 55.4.

`Best_fields` and `most_fields` types are field centric so it should return only min match of 2 per field.
https://www.elastic.co/guide/en/elasticsearch/reference/2.3/query-dsl-multi-match-query.html

Is minimum_should_match working in docker 55.4 or 56.0 ?
Any advice on how to change the clause to a working one with  ?

**Comment by joemoe** [comment]
hi @joalmjoalm, we'll check this and give you some feedback.

**Comment by chaudum** [comment]
hi @joalmjoalm 
I'm investigating the problem. It seems that the `minimum_should_match` option was ignored in boolean queries (both in `0.55` and `0.56`).
We'll provide a fix asap!

**Comment by chaudum** [comment]
@joalmjoalm 
the bug is fixed and will be available with the next releases of `0.55` and `0.56`

**Comment by joalmjoalm** [comment]
Great,
Many thanks.
